pkg/metrics/metrics.go

```diff
@@ -0,0 +1,218 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package metrics
```

Comment:
How is the prom client testing support? Is it possible to write tests that check that samples were recorded with certain labels set?

Reply:
It is, but it is a bit of a string matching exercise. Happy to stick something in if you are keen...

https://github.com/jetstack/cert-manager/blob/master/pkg/metrics/certificates_test.go

Reply:
I don't think it's critical, but I always prefer tests if they aren't too onerous.